src/main/java/com/google/sps/servlets/EditEventServlet.java

```diff
@@ -0,0 +1,304 @@
+        eventEntity.setProperty(Constants.GOING_ATTENDEES_PARAM, goingAttendees);
+        eventEntity.setProperty(Constants.NOT_GOING_ATTENDEES_PARAM, notGoingAttendees);
+        eventEntity.setProperty(Constants.CREATOR_PARAM, currentUserID);
+
+        datastore.put(eventEntity);
+        break;
+      }
+    }
+    json.put("success", "true");
+    json.put("bad-time", "false");
+  }
+
+/**
+  if event is finished in one day, date display should be:
+      EEE MMM dd, yyyy, startTime - endTime
+  if the event ends on a different day than it starts:
+      EEE MM Mdd, yyyy, startTime - EEE MMM dd, yyyy, endTime
+
+*/
+  private String createDateTimeDisplay(Date startDate, String startTime, Date endDate,
+                                String endTime, String timeZone) {
+    String dateTime = "";
+    String dateString;
+    SimpleDateFormat formatter = new SimpleDateFormat("EEE MMM dd, yyyy"); 
+
+    if (startDate.equals(endDate)) {
+      dateString = formatter.format(startDate);
+      dateTime += dateString + ", " + startTime + " - " + endTime + " " + 
+                  timeZone;
+    }
+    else {
+      String startDateString = formatter.format(startDate);
+      String endDateString = formatter.format(endDate);
+      dateTime += startDateString + ", " + startTime + " - " + endDateString;
+      dateTime += ", " + endTime + " " + timeZone;
+    }
+
+    return dateTime;
+  }
+
+/** Make sure the inputted dates and times are valid. */
+  private boolean verifyDateTimes (Date startDate, Date endDate, JSONObject json) {
+    if (endDate.before(startDate) || endDate.equals(startDate)) {
+      json.put("bad-time", "true");
+      return false;
+    }
+    return true;
+  }
+
+/**
+  Gets start and end times for display purposes -- converts times after 12 to standard
+  as opposed to military (i.e 13 to 1)
+*/
+  private String getTimeDisplay(String time) {
+    String oldHour;
+    int hour;
+    String hourForDisplay;
+    String period;
+    int firstChar = Integer.parseInt(time.substring(0, 1));
+    int firstTwoChars = Integer.parseInt(time.substring(0, 2));
+
+    if (firstChar == 0 && firstTwoChars != 00) {
+      hourForDisplay = time.substring(1, 2);
+      period = "am";
+    }
+    else if (firstTwoChars == 00) {
+      hourForDisplay = "12";
+      period = "am";
+    }
+    else if (firstTwoChars == 10 || firstTwoChars == 11) {
+      hourForDisplay = String.valueOf(firstTwoChars);
+      period = "am";
+    }
+    else if (firstTwoChars == 12) {
+      hourForDisplay = String.valueOf(firstTwoChars);
+      period = "pm";
+    }
+    else {
+      oldHour = time.substring(0, 2);
+      hour = Integer.parseInt(oldHour);
+      int hourForDisplayInt = hour - 12;
+      hourForDisplay = String.valueOf(hourForDisplayInt);
+      period = "pm";
+    }
+
+    String min = time.substring(3);
+    hourForDisplay += ":" + min + period;
+
+    return hourForDisplay;
```

Comment:
Consider instead parsing the string to a date/time object, and then formatting with SimpleDateFormat.

```
return new SimpleDateFormat("hh:mm a").format(dateTime);
```

https://docs.oracle.com/javase/tutorial/i18n/format/simpleDateFormat.html